index.js

```diff
@@ -347,7 +352,7 @@ Router.prototype.process_params = function process_params(layer, called, req, re
   var params = this.params
 
   // captured parameters from the layer, keys and values
-  var keys = layer.keys
+  var keys = layer.matchedPath ? layer.matchedPath.keys : []
```

Comment:
Probably creating an empty array here is unnecessary object allocation when the next line will accept `keys` being a falsy-value.

Reply:
Updated to `keys = layer.matchedPath && layer.matchedPath.keys`